File: src/main/java/org/bitecodelabs/botlerdaemon/BotlerDaemon.java
package org.bitecodelabs.botlerdaemon;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents;
import org.bitecodelabs.botlerdaemon.config.Config;
import org.bitecodelabs.botlerdaemon.event.PlayerLoginHandler;
import org.bitecodelabs.botlerdaemon.event.PlayerLogoutHandler;
import org.bitecodelabs.botlerdaemon.websocket.SocketClient;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class BotlerDaemon implements ModInitializer {

    public static final String MOD_ID = "Botler Daemon";
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);
    @Override
    public void onInitialize() {

        BotlerDaemon.LOGGER.info("Starting Botler Daemon");
        BotlerDaemon.LOGGER.info(Config.BOTLER_WEBSOCKET_HOST);

        SocketClient socketClient = SocketClient.getInstance();

        socketClient.emitEvent(String.valueOf(SocketClient.Events.BOTLER_SERVER_START), Config.BOTLER_SERVER_NAME);

        ServerPlayConnectionEvents.JOIN.register((new PlayerLoginHandler()));

        ServerPlayConnectionEvents.DISCONNECT.register((new PlayerLogoutHandler()));

    }
}

File: src/main/java/org/bitecodelabs/botlerdaemon/BotlerDaemon.kt
package org.bitecodelabs.botlerdaemon

import net.fabricmc.api.DedicatedServerModInitializer
import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents
import org.bitecodelabs.botlerdaemon.connections.SocketClient
import org.bitecodelabs.botlerdaemon.events.PlayerLoginHandler
import org.bitecodelabs.botlerdaemon.events.PlayerLogoutHandler
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class BotlerDaemon : DedicatedServerModInitializer {

    private val socketClient = SocketClient()
    override fun onInitializeServer() {
        LOGGER.info("Starting Botler Daemon")
        socketClient.setup()
        ServerPlayConnectionEvents.JOIN.register(PlayerLoginHandler())
        ServerPlayConnectionEvents.DISCONNECT.register(PlayerLogoutHandler())
    }

    companion object {
        private const val MOD_ID = "Botler Daemon"
        @JvmField
        var LOGGER: Logger = LoggerFactory.getLogger(MOD_ID)
    }
}